app/controllers/items_controller.rb

```diff
@@ -22,6 +22,28 @@ def show
     end
   end
   
+  def edit
+    @item = Item.find(params[:id])
+  end
+  
+  def update
+    @item = Item.find(params[:id])
+    if (current_user.items.include?(@item) and @item.update_attributes(item_params))
+      redirect_to item_path (@item), :notice => "#{@item.name} updated."
+    else
+      flash[:alert] = "#{@item.name} could not be updated: " + @item.errors.full_messages.join(",")
+      render 'edit'
+    end
+  end
+  
+  def destroy
+    @item = Item.find(params[:id])
+    if (current_user.items.include?(@item))
+      @item.destroy
+      redirect_to items_path, :notice => "#{@item.name} deleted."
```

Comment:
will @item.name cause an error since the item has  been deleted?

Reply:
that's a great question - I don't quite know the details of this, but rails actually handles this internally ( i think by storing the properties of item until the function returns)

Reply:
So my short answer is no, I don't think so.